File: typescript/src/Wall/WallDAO.ts
import {Brick} from './Brick';
import {NotTestableCallError} from '../Error/NotTestableCallError';
import {User} from '../User/User';
import {WallDAOInterface} from './WallDAOInterface';

export class WallDAO implements WallDAOInterface {
    /**
     * @deprecated
     */
    static addBrickToUser(user: User, brick: Brick): void {
        throw new NotTestableCallError('WallDAO should not be invoked on an unit test.');
    }

    /**
     * @deprecated
     */
    static findBricksByUser(user: User): Brick[] {
        throw new NotTestableCallError('WallDAO should not be invoked on an unit test.');
    }

    getBricks(user: User): Brick[] {
        return WallDAO.findBricksByUser(user);
    }

    addBrick(user: User, brick: Brick): void {
        WallDAO.addBrickToUser(user, brick);
    }
}

File: dotNet/src/Wall/WallDAO.cs
using System.Collections.Immutable;
using src.Exception;

namespace src.Wall;

public class WallDAO : IWallDAO
{
    [Obsolete]
    public static List<Brick> FindBricksByUser(src.User.User user)
    {
        throw new NotTestableCallException("WallDAO should not be invoked on an unit test.");
    }

    [Obsolete]
    public static void AddBrickToUser(src.User.User user, Brick brick)
    {
        throw new NotTestableCallException("WallDAO should not be invoked on an unit test.");
    }

    public ImmutableArray<Brick> GetBricks(User.User user)
    {
        return ImmutableArray.ToImmutableArray(FindBricksByUser(user));
    }

    public void AddBrick(User.User user, Brick brick)
    {
        AddBrickToUser(user, brick);
    }
}
